src/pages/searching.tsx

```diff
@@ -98,6 +98,14 @@ const Searching = () => {
     };
   }, [app.currentChatId, authState.email, authState.loginId, router, socket]);
 
+  useEffect(() => {
+    if (!app.currentChatId) {
+      return;
+    }
+
+    router.push('/anonymous')
```

Comment:
This could cause a problem tho, why not just let the user go back themselves by clicking a button like we previously did

Reply:
Because when the chat is restored, `searching` page does not stop the animation and the user does not have access to the chat interface. They never know when the chat is restored.

Reply:
hold up, this is to work when the user is already searching and not before? I think before is best right?

Reply:
yep, that's the flow

you're saying even if the user is on the homepage, it redirects back?

Reply:
> you're saying even if the user is on the homepage, it redirects back?

No. User gets redirected only if they're on `/searching`

Reply:
I think it's better if we:
- check if the user has a chatid and prevent them from going to the searching page in general
- only redirect them outside of the chat page

Reply:
I think we're already doing that.
There's no better place to redirect users from `searching` page than `searching` page itself. We could use a middleware, but it seems middlewares in Next only run on server, so checking for the currentChat Id is quite tricky due to our current structure.

Reply:
it's in the context no?

Reply:
but i also think both places should have it, the first one to check in the home page and the second to check in the searching page.

You said the first one is done already right?

Reply:
Second one is already done.

But, is the the first not going to disturb? I mean we could display other stuff there instead of just redirecting.

Reply:
I think the first way is dome the so nevermind actually